backend/api/professor/cohorts.py

```
@@ -415,6 +478,60 @@ async def add_student_to_cohort(
         approved_at=cohort_student.approved_at
     )
 
+@router.put("/{cohort_unique_id}/students/{student_id}", response_model=CohortStudentResponse)
+async def update_student_enrollment(
+    cohort_unique_id: str,
+    student_id: int,
+    student_data: CohortStudentUpdate,
+    current_user: User = Depends(get_current_user),
```

Ruff (0.13.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)